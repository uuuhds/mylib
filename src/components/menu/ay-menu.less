.ay {
    .ant-menu {
        background-color: transparent;
    }
    .ant-menu-horizontal > .ant-menu-item a {
        color: #fff;
    }
    .ant-menu-title-content a {
        color: #fff;
    }
    .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-submenu:hover::after {
        border-color: transparent;
    }
    .ant-menu-item {
        background: transparent;
        &.ant-menu-item-selected,
        &.ant-menu-item-active {
            background: rgba(0, 0, 0, 0.1);
            border-radius: 4px;
            &::after {
                border-bottom-color: transparent;
            }
        }
    }
    .ant-menu-submenu {
        &.ant-menu-submenu-active,
        &.ant-menu-submenu-selected {
            background: rgba(0, 0, 0, 0.1);
            border-radius: 4px;
            &::after {
                border-bottom-color: transparent;
            }
        }
    }

    .ant-menu-horizontal {
        border-bottom: none;
    }
    .ant-menu-title-content {
        color: #fff;
    }
    .ant-menu-horizontal:not(.ant-menu-dark) > .ant-menu-item-active::after {
        border-color: transparent;
    }
    .redIcon {
        height: 8px;
        width: 8px;
        background-color: #e41010;
        border-radius: 50%;
        position: absolute;
        right: 10px;
        top: 16px;
    }
}

.ant-menu-submenu-popup ul {
    width: 20px;
}

.ay-sub-menu {
    .ant-menu {
        border-radius: 4px;
        &.ant-menu-vertical.ant-menu-sub {
            min-width: 120px;
        }
        li {
            width: 120px;
            text-align: center;
        }
    }
}
