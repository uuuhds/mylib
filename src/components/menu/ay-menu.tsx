import React from 'react';
import { Menu, MenuProps } from 'antd';
import './ay-menu.less';

export type AyMenuProps = MenuProps;

export const HorizontalMenu = ({ children, ...rest }: AyMenuProps) => {
  return (
    <Menu mode="horizontal" {...rest}>
      {children}
    </Menu>
  );
};
