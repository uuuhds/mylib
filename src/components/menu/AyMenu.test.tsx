import React from 'react';
import renderer, { act } from 'react-test-renderer';
// import '@testing-library/jest-dom/extend-expect';
import { DefaultMenu } from './AyMenu.stories';
import { AyMenuProps } from './ay-menu';

// const delay = async (miliseconds = 0) => {
//   await new Promise(resolve => {
//     setTimeout(resolve, miliseconds);
//   });
// };

it('renders ay menu', async () => {
  const props = DefaultMenu.args as AyMenuProps;
  await act(async () => {
    const component = await renderer.create(<DefaultMenu {...props} />);
    let tree = component.toJSON();
    // await delay(2000);
    // console.log('tree', tree);
    expect(tree).toMatchSnapshot();
  });
});
