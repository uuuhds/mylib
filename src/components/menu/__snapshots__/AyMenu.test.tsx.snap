// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ay menu 1`] = `
<div
  className="ay"
  style={
    Object {
      "alignItems": "center",
      "background": "linear-gradient(180deg, #4389f1 0%, rgba(67, 137, 241, 0.7) 100%)",
      "display": "flex",
      "height": 80,
    }
  }
>
  <ul
    className="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light"
    data-menu-list={true}
    onKeyDown={[Function]}
    role="menu"
    tabIndex={0}
  >
    <li
      className="ant-menu-overflow-item ant-menu-item ant-menu-item-selected ant-menu-item-only-child"
      data-menu-id="rc-menu-uuid-test-mail"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      path=""
      role="menuitem"
      style={
        Object {
          "height": undefined,
          "opacity": 1,
          "order": 0,
          "overflowY": undefined,
          "pointerEvents": undefined,
          "position": undefined,
        }
      }
      tabIndex={-1}
    >
      <span
        className="ant-menu-title-content"
      >
        订单管理
      </span>
    </li>
    <li
      aria-disabled={true}
      className="ant-menu-overflow-item ant-menu-item ant-menu-item-disabled ant-menu-item-only-child"
      data-menu-id={null}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      path=""
      role="menuitem"
      style={
        Object {
          "height": undefined,
          "opacity": 1,
          "order": 1,
          "overflowY": undefined,
          "pointerEvents": undefined,
          "position": undefined,
        }
      }
      tabIndex={null}
    >
      <span
        className="ant-menu-title-content"
      >
        禁用菜单
      </span>
    </li>
    <li
      className="ant-menu-overflow-item ant-menu-submenu ant-menu-submenu-horizontal"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      redIcon={true}
      role="none"
      style={
        Object {
          "height": undefined,
          "opacity": 1,
          "order": 2,
          "overflowY": undefined,
          "pointerEvents": undefined,
          "position": undefined,
        }
      }
    >
      <div
        aria-controls="rc-menu-uuid-test-SubMenu-popup"
        aria-expanded={false}
        aria-haspopup={true}
        className="ant-menu-submenu-title"
        data-menu-id={null}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={null}
        tabIndex={-1}
        title={null}
      >
        <span
          className="ant-menu-title-content"
        >
          批量打印
        </span>
        <i
          className="ant-menu-submenu-arrow"
        />
      </div>
    </li>
    <li
      aria-hidden={true}
      className="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      role="none"
      style={
        Object {
          "height": 0,
          "opacity": 0,
          "order": 9007199254740991,
          "overflowY": "hidden",
          "pointerEvents": "none",
          "position": "absolute",
        }
      }
    >
      <div
        aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
        aria-expanded={false}
        aria-haspopup={true}
        className="ant-menu-submenu-title"
        data-menu-id="rc-menu-uuid-test-rc-menu-more"
        onClick={[Function]}
        onFocus={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="menuitem"
        style={null}
        tabIndex={-1}
        title={null}
      >
        <span
          aria-label="ellipsis"
          className="anticon anticon-ellipsis"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="ellipsis"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
            />
          </svg>
        </span>
        <i
          className="ant-menu-submenu-arrow"
        />
      </div>
    </li>
  </ul>
  <div
    aria-hidden={true}
    style={
      Object {
        "display": "none",
      }
    }
  />
</div>
`;
